lib/bolt/pal.rb

```diff
@@ -157,7 +171,7 @@ def detect_project_conflict(project, environment)
     def in_bolt_compiler
       # TODO: If we always call this inside a bolt_executor we can remove this here
       setup
-      r = Puppet::Pal.in_tmp_environment('bolt', modulepath: @modulepath, facts: {}) do |pal|
+      r = Puppet::Pal.in_tmp_environment('bolt', modulepath: @modulepath.bolt_compiler_modulepath, facts: {}) do |pal|
```

Comment:
why not just use the `modulepath` method? https://github.com/puppetlabs/bolt/pull/2252/files#diff-6f471e6eed4bf0e0390c0a6a5aa5304cfd0d7e22a034795ea51c18da15381b90R79